lib/r10k/api/git.rb

```diff
@@ -17,7 +19,13 @@ def clean(opts={})
       end
 
       def rev_parse(rev, opts={})
-        provider.rev_parse(rev, opts)
+        result = provider.rev_parse(rev, opts)
+
+        if result.success?
+          return result.stdout.strip
+        else
+          raise CommandFailedError.new(result.stderr)
+        end
```

Comment:
This is the part that is not going to work with the jgit provider. Need to sit down and decide how we want to standardize the return values from the various providers.